source/base/data_out_base.cc

```diff
@@ -850,11 +850,6 @@ namespace
      * technique.
      */
     std::ostream &stream;
-
-    /**
-     * The flags controlling the output
-     */
-    const DataOutBase::GmvFlags flags;
   };
```

Comment:
Do you also get this warning if you leave the assignment to the variable in the constructor, or only if you remove the assignment but leave the variable declaration? It would seem to be that assigning the variable in the constructor -- even if the variable is an empty struct -- should constitute a "use" of it.

If you do find that you need to remove the variable to avoid the warning, please add code such as 
```
  // we removed the storing the GmvFlags structure because it led to errors
  // when it was empty. however, storing it should be reenabled if the structure
  // ever gains any member fields -- the following test verifies that the structure
  // at least has no fields that are larger than a single byte
  Assert (sizeof(GmvFlags)==1, ExcNotImplemented());
```